tests/test_suite.py

```diff
@@ -8,51 +8,18 @@ class TestSuite(unittest.TestCase):
     maxDiff = None
     DIR = os.path.join(os.path.dirname(__file__), "suite")
 
-    def _test_file(self, basename):
-        with open(os.path.join(self.DIR, f"{basename}.in"), "r") as f:
-            actual_input = f.read()
-
-        try:
-            with open(os.path.join(self.DIR, f"{basename}.out"), "r") as f:
-                expected_output = f.read()
-        except FileNotFoundError:
-            expected_output = None
-
-        try:
-            with open(os.path.join(self.DIR, f"{basename}.err"), "r") as f:
-                expected_error = f.read().strip()
-        except FileNotFoundError:
-            expected_error = None
-
-        if expected_output is None and expected_error is None:
-            self.fail(f"{basename} has no expected error or output.")
-
-        if expected_output is not None and expected_error is not None:
-            self.fail(f"{basename} has both expected error and output.")
-
-        try:
-            actual_output = DjHTML(actual_input).indent(4)
-            actual_error = None
-        except Exception as err:
-            actual_error = str(err).strip()
-            actual_output = None
-
-        if actual_error is not None and expected_error is None:
-            self.fail(f"Unexpected error: {actual_error}")
-        if actual_error is None and expected_error is not None:
-            self.fail(f"Unexpected success, expected error: {expected_error}")
-
-        self.assertEqual(actual_output, expected_output)
-        self.assertEqual(actual_error, expected_error)
-
     def test_available_files(self):
         """
         Loop over all the files in the suite directory and compare the
         expected output to the actual output.
 
         """
         for filename in os.listdir(self.DIR):
-            if filename.endswith(".in"):
-                basename, _ = os.path.splitext(filename)
-                with self.subTest(basename):
-                    self._test_file(basename)
+            with self.subTest(filename):
+                self._test_file(filename)
+
+    def _test_file(self, filename):
+        with open(os.path.join(self.DIR, filename)) as f:
+            expected_output = f.read()
+        actual_output = DjHTML(expected_output).indent(4)
```

Comment:
The current test-suite would still pass if we replace `djhtml` with `cat`. So I believe that this change in test-suite is a degradation in quality assurance.

(Though: an idempotency-check inside DjHTML itself would already be an excellent improvement! So if `djhtml(djhtml(text)) != djhtml(text)`, ask the user to create a bug-report.)

Reply:
Even better, I've added a `DjHTML(expected_output).indent(0)` to remove all leading (and trailing) spaces from the expected output before indenting it to 4, with an `assertNotEqual()` in between. The "cat test" would now fail, and idempotency is (somewhat) guaranteed!